packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -259,6 +269,69 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
   return self;
 }
 
+- (void)setupPipController {
+  if ([AVPictureInPictureController isPictureInPictureSupported]) {
+    self.pictureInPictureController =
+        [[AVPictureInPictureController alloc] initWithPlayerLayer:self.playerLayer];
+    [self setAutomaticallyStartPictureInPicture:NO];
+    self.pictureInPictureController.delegate = self;
+  }
+}
+
+- (void)setAutomaticallyStartPictureInPicture:
+    (BOOL)canStartPictureInPictureAutomaticallyFromInline {
+  if (!self.pictureInPictureController) return;
+  if (@available(iOS 14.2, *)) {
+    self.pictureInPictureController.canStartPictureInPictureAutomaticallyFromInline =
+        canStartPictureInPictureAutomaticallyFromInline;
+  }
+}
+
+- (void)setPictureInPictureOverlayRect:(CGRect)frame {
+  if (_player) {
+    self.playerLayer.frame = frame;
+  }
+}
+
+- (void)startOrStopPictureInPicture:(BOOL)shouldPictureInPictureStart {
+  if (self.isPictureInPictureStarted == shouldPictureInPictureStart) {
```

Comment:
Should this also early return if `pictureInPictureController` is nil?  It seems like a mistake to ever set `isPictureInPictureStarted` to `YES` when `isPictureInPictureSupported` isn't true.

Reply:
Yes you are right. Fixed this